internal/apps/user/projector.go

```
@@ -68,9 +67,34 @@ func GetUserName(ds projector.Datastore, uid int) (string, error) {
 		return user.Username, nil
 	}
 
-	if user.Level != "" {
-		parts = append(parts, fmt.Sprintf("(%s)", user.Level))
+	return strings.Join(parts, " "), nil
+}
+
+// GetUserLevel returns the structre level for user id.
+func GetUserLevel(ds projector.Datastore, uid int) (string, error) {
+	var user struct {
+		Level string `json:"structure_level"`
 	}
 
-	return strings.Join(parts, " "), nil
+	if err := ds.Get("users/user", uid, &user); err != nil {
+		return "", projector.NewClientError("users/user with id %d does not exist", uid)
+	}
+
+	return user.Level, nil
+}
+
+// GetUserName returns the display name for an user id.
+func GetUserName(ds projector.Datastore, uid int) (string, error) {
+	shotName, err := GetUserShortName(ds, uid)
+	if err != nil {
+		return "", err
```

You should Return a new error with

fmt.Errorf("getting short name: %w", err)